File: src/main/java/com/jiro4989/tkfm/model/ImageFormatModel.java
package com.jiro4989.tkfm.model;

import javafx.beans.property.*;

/** 画像フォーマット。画像フォーマット名、行数、列数、1タイルあたりの矩形を管理する。 */
public class ImageFormatModel {
  /** フォーマットの名前 */
  private final String name;
  /** 行数 */
  private final IntegerProperty row;
  /** 列数 */
  private final IntegerProperty col;
  /** 1タイルあたりの矩形 */
  private final Rectangle rect;

  public ImageFormatModel(String name, int row, int col, Rectangle rect) {
    this.name = name;
    this.row = new SimpleIntegerProperty(row);
    this.col = new SimpleIntegerProperty(col);
    this.rect = rect;
  }

  public IntegerProperty rowProperty() {
    return row;
  }

  public IntegerProperty colProperty() {
    return col;
  }

  public String getName() {
    return name;
  }

  public Rectangle getRectangle() {
    return rect;
  }
}

File: src/main/kotlin/com/jiro4989/tkfm/model/ImageFormatModel.kt
package com.jiro4989.tkfm.model

import javafx.beans.property.IntegerProperty
import javafx.beans.property.SimpleIntegerProperty

/** 画像フォーマット。画像フォーマット名、行数、列数、1タイルあたりの矩形を管理する。 */
data class ImageFormatModel(
    /** フォーマットの名前 */
    val name: String,
    /** 行数 */
    val rowProperty: IntegerProperty,
    /** 列数 */
    val colProperty: IntegerProperty,
    /** 1タイルあたりの矩形 */
    val rectangle: RectangleModel
) {
  var row
    get() = rowProperty.get()
    set(value) = rowProperty.set(value)
  var col
    get() = colProperty.get()
    set(value) = colProperty.set(value)

  constructor(name: String, row: Int, col: Int, rect: RectangleModel) : this(
      name, SimpleIntegerProperty(row), SimpleIntegerProperty(col), rect)
}
